# Day 3 Workout - Types and Structures

In [1]:
student = "Sammy"
grade = 88.444

What is the type of student?

In [7]:
type(student)

str

What is the type of the first character of student?

In [8]:
type(student[0])

str

Concatenate the variables containing student name and grade.

The output should be formatted to be `The grade of Sammy is 88.444%`

In [2]:
## Method 1
"The grade of " + student + " is " + str(grade) + "%"

'The grade of Sammy is 88.444%'

In [3]:
## Method 2
f"The grade of {student} is {grade}%"

'The grade of Sammy is 88.444%'

In [4]:
## Method 3
"The grade of {0} is {1}%".format(student, grade)

'The grade of Sammy is 88.444%'

Convert `grade` to `int`, re-format the variables.

The output should be formatted to be `The grade of Sammy is 88%`

In [5]:
## Method 1
"The grade of " + student + " is " + str(int(grade)) + "%"

'The grade of Sammy is 88%'

In [6]:
f"The grade of {student} is {grade:.0f}%"

'The grade of Sammy is 88%'

Concatenate the variables containing student name and grade.

Keep only one decimal digit.

The output should be formatted to be `The grade of Sammy is 88.4%`

In [10]:
## Method 1
rounded_grade = round(grade, 1)
f"The grade of {student} is {rounded_grade}%"

'The grade of Sammy is 88.4%'

In [11]:
## Method 2
f"The grade of {student} is {grade:.1f}%"

'The grade of Sammy is 88.4%'

Using string functions, convert `msg` to:
- upper case
- title case
- capitalize

In [12]:
msg = "data analytics is the future"

In [13]:
msg.upper()

'DATA ANALYTICS IS THE FUTURE'

In [14]:
msg.title()

'Data Analytics Is The Future'

In [15]:
msg.capitalize()

'Data analytics is the future'

Extract the email address in the new string `msg`

In [17]:
msg = """My name is James Bond and my email address
is james.bond@mi6.co.uk
Do not hesitate to write me. I'm a man for all seasons
"""

In [18]:
## Method 1

# convert new line to a space
msg2 = msg.replace('\n', ' ')
# first locate the location of the @
at_index = msg2.find('@')
# next find the white space after the @
space_right = msg2.find(' ', at_index)
# next find the white space before the @
space_left = msg2.rfind(' ', 0, at_index)

bond_email=msg2[space_left+1:space_right]
bond_email

'james.bond@mi6.co.uk'

In [19]:
## Method 2
words = msg.split()

bond_email = None
for word in words:
    if '@' in word:
        bond_email = word
bond_email

'james.bond@mi6.co.uk'